src/frequenz/channels/util/_timer.py

```diff
@@ -652,10 +652,15 @@ async def ready(self) -> bool:
 
         now = self._now()
         time_to_next_tick = self._next_tick_time - now
+
         # If we didn't reach the tick yet, sleep until we do.
-        if time_to_next_tick > 0:
+        # We need to do this in a loop also reacting to the reset event, as the timer
+        # could be reset while we are sleeping, in which case we need to recalculated
```

Comment:
I thought this was amended in the original patch :(
```suggestion
        # could be reset while we are sleeping, in which case we need to recalculate
```

Reply:
Damn, this is in the original PR too #241.